wp-content/themes/core/assets/pcss/layout/default.pcss

```diff
@@ -0,0 +1,33 @@
+/* -------------------------------------------------------------------------
+ *
+ * Layouts
+ *
+ * ------------------------------------------------------------------------- */
+
+/* -------------------------------------------------------------------------
+ *
+ * Layout: Align Wide
+ *
+ * Adjustments for fluid value in theme.json settings.layout.wideSize
+ *
+ * ------------------------------------------------------------------------- */
+
+/*
+ * Max-width of blocks aligned wide should not account for grid margin
+ * when nested in .has-global-padding because the container applies
+ * grid margin as padding.
+ */
+.has-global-padding.is-layout-constrained > .alignwide,
+:is(.editor-styles-wrapper) .is-root-container.has-global-padding.is-layout-constrained > .alignwide {
+	max-width: 100%;
+}
+
+/*
+ * Max-width of blocks aligned wide nested inside two .has-global-padding
+ * containers need to re-account for grid-margin.
+ * Example: Columns block aligned wide inside group block aligned full.
+ */
+.has-global-padding :is(.has-global-padding) > .alignwide,
+:is(.editor-styles-wrapper) .has-global-padding :is(.has-global-padding) > .alignwide {
+	max-width: var(--wp--style--global--wide-size);
+}
```

Comment:
A few questions & suggestions here:
* I wonder if we need a bit more explanation of these conditions. I'm fairly comfortable with the editor, and I'm not sure I follow the complexity happening here.
* If this is caused by other styles (WP Core maybe?) is there an alternative where we unset whatever those styles are to then create the conditions we want instead?
* I wonder if we move this (or rename it) to an alignments partial and then make  `.alignwide` a top-level selector with the conditional as nested parents? I'm not sure we'll really need/have a "default" alignment system any more since the block editor technically blew that all up.

Reply:
@dpellenwood I added a bit more context to the comments here. I did have to play with it a bit to understand what's going on but essentially because we're using `calc(100% - var(--wp--custom--spacing--grid-margin) * 2)` as the value for `alignwide` instead of a set pixel value, we need to override cases where the padding on the left/right of blocks would be doubled. The first block is overriding one of these cases, and the second is overriding the first, essentially.